nengo/builder/operator.py

```diff
@@ -388,7 +388,10 @@ def make_step(self, signals, dt, rng):
 
         if inc:
             def step_copy():
-                dst[dst_slice] += src[src_slice]
+                if type(dst_slice) == type(Ellipsis):
+                    dst[dst_slice] += src[src_slice]
+                else:
+                    np.add.at(dst, dst_slice, src[src_slice])
         else:
             def step_copy():
                 dst[dst_slice] = src[src_slice]
```

Comment:
In this case, if there are repeated indices in ``dst_slice``, then the behaviour is kind of undefined. I think we should have an assertion here to make sure that's not the case. I don't think a user should ever be able to trip that assertion, but it could save a developer some trouble.